lib/rules/object-shorthand.js

```diff
@@ -285,17 +285,11 @@ module.exports = {
             const fnBody = sourceCode.text.slice(arrowToken.range[1], node.value.range[1]);
 
             let shouldAddParensAroundParameters = false;
-            let tokenBeforeParams;
-
-            if (node.value.params.length === 0) {
-                tokenBeforeParams = sourceCode.getFirstToken(node.value, astUtils.isOpeningParenToken);
-            } else {
-                tokenBeforeParams = sourceCode.getTokenBefore(node.value.params[0]);
-            }
+            const tokenBeforeParams = sourceCode.getFirstToken(node.value, token => astUtils.isOpeningAngleToken(token) || astUtils.isOpeningParenToken(token));
```

Comment:
can we not explicitly check `<`/`>`(it's a ts-specific token that eslint not officially supported)?

possible just find the first token after `:`?

Reply:
Perhaps considering this special case? `{ case: ( async () => {}) }`
I followed the original logic. If the `<>` token is removed, it may require some changes to the original logic. Is that okay?

Reply:
good point! The easiest way I've seen to do this is to check if the character is expected, e.g `function`/`async`, if not we can return `null`, so the code won't be fixed. wdyt?

Reply:
But this case `{ case: ( async () => {}) }` should be fixed as `{ async  case() {} }` in the original logic.
And the original logic used `methodPrefix` for `async` or `*`, instead of directly using `async () => {}` after the colon.

Reply:
Well, given the complexity of the change, I'm fine with the current one. 😄

Reply:
But I think the easiest way is to use `node.value.typeParameters` directly，I will show u.